bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowerMaxCommDriver.java

```diff
@@ -0,0 +1,621 @@
+        } else {
+            int val = (device == null) ? 1 : (1 << device);
+            byte[] dynPart = new byte[3];
+            dynPart[0] = code;
+            dynPart[1] = (byte) (val & 0x000000FF);
+            dynPart[2] = (byte) (val >> 8);
+
+            done = sendMessage(new PowerMaxBaseMessage(PowerMaxSendType.X10PGM, dynPart), false, 0);
+        }
+        return done;
+    }
+
+    /**
+     * Send a message to the PowerMax alarm panel to bypass a zone or to not bypass a zone
+     *
+     * @param bypass
+     *            true to bypass the zone; false to not bypass the zone
+     * @param zone
+     *            the zone number (first zone is number 1)
+     * @param pinCode
+     *            the PIN code. A string of 4 characters is expected
+     *
+     * @return true if the message was sent or false if not
+     */
+    public boolean sendZoneBypass(boolean bypass, byte zone, String pinCode) {
+        logger.debug("sendZoneBypass(): bypass = {}, zone = {}", bypass ? "true" : "false", zone);
+
+        boolean done = false;
+
+        if ((pinCode == null) || (pinCode.length() != 4)) {
+            logger.warn("PowerMax alarm binding: zone bypass rejected due to invalid PIN code");
+        } else if ((zone < 1) || (zone > PowerMaxPanelSettings.getThePanelSettings().getNbZones())) {
+            logger.warn("PowerMax alarm binding: invalid zone number: {}", zone);
+        } else {
+            try {
+                int val = (1 << (zone - 1));
+
+                byte[] dynPart = new byte[10];
+                dynPart[0] = (byte) Integer.parseInt(pinCode.substring(0, 2), 16);
+                dynPart[1] = (byte) Integer.parseInt(pinCode.substring(2, 4), 16);
+                int i;
+                for (i = 2; i < 10; i++) {
+                    dynPart[i] = 0;
+                }
+                i = bypass ? 2 : 6;
+                dynPart[i++] = (byte) (val & 0x000000FF);
+                dynPart[i++] = (byte) ((val >> 8) & 0x000000FF);
+                dynPart[i++] = (byte) ((val >> 16) & 0x000000FF);
+                dynPart[i++] = (byte) ((val >> 24) & 0x000000FF);
+
+                done = sendMessage(new PowerMaxBaseMessage(PowerMaxSendType.BYPASS, dynPart), false, 0);
+                if (done) {
+                    done = sendMessage(new PowerMaxBaseMessage(PowerMaxSendType.BYPASSTAT), false, 0);
+                }
+            } catch (NumberFormatException e) {
+                logger.warn("PowerMax alarm binding: zone bypass rejected due to invalid PIN code");
+            }
+        }
+        return done;
+    }
+
+    /**
+     * Send a message to set the alarm time and date using the system time and date
+     *
+     * @return true if the message was sent or false if not
+     */
+    public boolean sendSetTime() {
+        logger.debug("sendSetTime()");
+
+        boolean done = false;
+
+        GregorianCalendar cal = new GregorianCalendar();
+        if (cal.get(Calendar.YEAR) >= 2000) {
```

Comment:
I need to provide a year from 2000 (16 for 2016). If current date is before year 2000, I just avoid synchronizing date/time.